packages/@ember/array/type-tests/mutable.ts

```diff
@@ -1,63 +1,67 @@
-import type MutableArray from '@ember/array/mutable';
-import { A } from '@ember/array';
+import MutableArray from '@ember/array/mutable';
 
 import { expectTypeOf } from 'expect-type';
 
-class Foo {}
+class Foo {
+  constructor(public name: string) {}
+}
 
-let foo = new Foo();
+let foo = new Foo('test');
 
-let arr = A([foo]);
+let originalArr = [foo];
+// This is not really the ideal way to set things up.
+MutableArray.apply(originalArr);
+let arr = originalArr as unknown as MutableArray<Foo>;
 
 expectTypeOf(arr).toMatchTypeOf<MutableArray<Foo>>();
 
 expectTypeOf(arr.replace(1, 1, [foo])).toEqualTypeOf<void>();
-// TODO: Why doesn't this fail?
+// @ts-expect-error invalid item
```

Comment:
It's nice that all of these fail now the way we want them to. 👍🏼

Reply:
I think we just needed a better test item :)